project-koku/koku
# COST-1067: Extending start_date query parameter to cover materialized view window

The `start_date` query parameter should be functional for the same number of billing months that we are storing in summary-materialized views.

This change will use the environment variable `RETAIN_NUM_MONTHS` to keep the summary data and API query validation in sync.

**Testing**
1. Set RETAIN_NUM_MONTHS=3 and verify 3 billing periods are accessible in a report API.
2. Verify that a date prior to the RETAIN_NUM_MONTHS=3 window throws a 400 error.
3. Change RETAIN_NUM_MONTHS=2 and repeat steps 1 and 2.

**Test Results**
[cost_1067_ut.txt](https://github.com/project-koku/koku/files/6010720/cost_1067_ut.txt)
